app/controllers/api/graphql_controller.rb

```diff
@@ -0,0 +1,83 @@
+module Api
+  class GraphqlController < ActionController::Base
+    include Foreman::ThreadSession::Cleaner
+    include Foreman::Controller::Timezone
+    include Foreman::Controller::RequireSsl
+    include Foreman::Controller::Session
+    include Foreman::Controller::Authentication
+
+    rescue_from Exception, :with => :generic_exception if Rails.env.production?
+
+    before_action :authenticate
+    around_action :set_timezone
+
+    def execute
+      result = if params[:_json]
+                 execute_multiplexed_graphql_query
+               else
+                 execute_single_graphql_query
+               end
+
+      render json: result
+    end
+
+    def api_request?
+      true
+    end
+
+    private
+
+    def execute_multiplexed_graphql_query
+      current_user = User.current
+      queries = params[:_json].map do |param|
+        {
+          query: param['query'],
+          operation_name: param['operationName'],
+          variables: ensure_hash(param['variables']),
+          context: {
+            current_user: current_user,
+            request_id: request.uuid
+          }
+        }
+      end
+      ForemanGraphqlSchema.multiplex(queries)
+    end
+
+    def execute_single_graphql_query
+      ForemanGraphqlSchema.execute(
+        params[:query],
+        variables: variables,
+        context: {
+          current_user: User.current,
+          request_id: request.uuid
+        }
+      )
+    end
+
+    def variables
+      ensure_hash(params[:variables])
+    end
+
+    def ensure_hash(ambiguous_param)
+      case ambiguous_param
+      when String
+        if ambiguous_param.present?
+          ensure_hash(JSON.parse(ambiguous_param))
```

Comment:
Should we re-raise `ArgumentError` when we get `JSON::ParserError` or handle it in some other way rather than leaving it to be rescued with `:generic_exception`?